classes/Dispatcher.php

```diff
@@ -1346,4 +1394,66 @@ public static function getControllersPhpselfList(string $dir, bool $base_name_ot
 
         return $controllersPhpself;
     }
+
+    /**
+     * Return id of entity matching rewrite
+     *
+     * @param string $rewrite
+     * @param string $controller
+     * @param string $routeName
+     *
+     * @return int the founded id
+     */
+    public function getIdFromRewrite($rewrite, $controller, $routeName): int
+    {
+        if (empty($rewrite) or empty($controller)) {
+            return 0;
+        }
+
+        $idLang = Context::getContext()->language->id;
+        $idShop = Context::getContext()->shop->id;
+
+        $query = new DbQuery();
+        $query->select('tl.id_' . $controller);
+        $query->from($controller . '_lang', 'tl');
+        $query->where('tl.`link_rewrite` = \'' . pSQL($rewrite) . '\'');
+        $query->where('tl.`id_lang` = ' . (int) $idLang);
+        $query->where('tl.`id_shop` = ' . (int) $idShop);
+        if ($controller == 'category') {
+            $query->innerJoin($controller, 't', 't.id_' . $controller . ' = tl.id_' . $controller);
+            $query->orderBy('t.active DESC, t.date_upd DESC');
+        }
```

Comment:
I don't understand why this idea still exists.
If you have these categories:
- Men (CAT 1) => Sportswear (CAT 4) => Jeans (CAT 7)
- Women (CAT 2) => Sportswear (CAT 5) => Jeans (CAT 8)
- Children (CAT 3) => Sportswear (CAT 6) => Jeans (CAT 9)

Your query on the rewritten URL "jeans" will always return the category 9 => $query->orderBy('t.active DESC, t.date_upd DESC'); which is not at all the expected result.
PrestaShop can only find an item by its ID or its direct parents.

Reply:
If the rewrite is the identifier, it has to be unique. 
This pr is a first step, if maintainers approve it, others improvements will be done.

This pr serve specific user cases of this optional feature : migration from another cms, or changing ids, ...

Reply:
BTW, you can set rewrite like this : men-sportswear-jeans mannually, or we could add a option for that